File: src/main/java/sk/hudak/prco/model/ErrorEntity.java
package sk.hudak.prco.model;

import sk.hudak.prco.api.ErrorType;
import sk.hudak.prco.api.EshopUuid;
import sk.hudak.prco.model.core.DbEntity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.SequenceGenerator;
import java.util.Objects;

@Entity(name = "ERROR")
public class ErrorEntity extends DbEntity {

    @Id
    @GeneratedValue(generator = "ERROR_SEC", strategy = GenerationType.SEQUENCE)
    @SequenceGenerator(name = "ERROR_SEC", sequenceName = "ERROR_SEC", allocationSize = 1)
    private Long id;

    @Enumerated(EnumType.STRING)
    private EshopUuid eshopUuid;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private ErrorType errorType;

    private String statusCode;

    private String message;

    @Column(length = 4000)
    private String fullMsg;

    private String url;

    private String additionalInfo;

    @Override
    public Long getId() {
        return id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }

    public EshopUuid getEshopUuid() {
        return eshopUuid;
    }

    public void setEshopUuid(EshopUuid eshopUuid) {
        this.eshopUuid = eshopUuid;
    }

    public ErrorType getErrorType() {
        return errorType;
    }

    public void setErrorType(ErrorType errorType) {
        this.errorType = errorType;
    }

    public String getStatusCode() {
        return statusCode;
    }

    public void setStatusCode(String statusCode) {
        this.statusCode = statusCode;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getFullMsg() {
        return fullMsg;
    }

    public void setFullMsg(String fullMsg) {
        this.fullMsg = fullMsg;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getAdditionalInfo() {
        return additionalInfo;
    }

    public void setAdditionalInfo(String additionalInfo) {
        this.additionalInfo = additionalInfo;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ErrorEntity that = (ErrorEntity) o;
        return Objects.equals(id, that.id) &&
                eshopUuid == that.eshopUuid &&
                errorType == that.errorType &&
                Objects.equals(statusCode, that.statusCode) &&
                Objects.equals(message, that.message) &&
                Objects.equals(fullMsg, that.fullMsg) &&
                Objects.equals(url, that.url) &&
                Objects.equals(additionalInfo, that.additionalInfo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, eshopUuid, errorType, statusCode, message, fullMsg, url, additionalInfo);
    }
}

File: src/main/java/sk/hudak/prco/model/ErrorEntity.kt
package sk.hudak.prco.model

import sk.hudak.prco.api.ErrorType
import sk.hudak.prco.api.EshopUuid
import sk.hudak.prco.model.core.DbEntity
import java.util.*
import javax.persistence.*

@Entity(name = "ERROR")
class ErrorEntity : DbEntity() {

    @Id
    @GeneratedValue(generator = "ERROR_SEC", strategy = GenerationType.SEQUENCE)
    @SequenceGenerator(name = "ERROR_SEC", sequenceName = "ERROR_SEC", allocationSize = 1)
    override var id: Long? = null

    @Enumerated(EnumType.STRING)
    var eshopUuid: EshopUuid? = null

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    var errorType: ErrorType? = null

    var statusCode: String? = null

    var message: String? = null

    @Column(length = 4000)
    var fullMsg: String? = null

    var url: String? = null

    var additionalInfo: String? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as ErrorEntity?
        return id == that!!.id &&
                eshopUuid == that.eshopUuid &&
                errorType === that.errorType &&
                statusCode == that.statusCode &&
                message == that.message &&
                fullMsg == that.fullMsg &&
                url == that.url &&
                additionalInfo == that.additionalInfo
    }

    override fun hashCode(): Int {
        return Objects.hash(id, eshopUuid, errorType, statusCode, message, fullMsg, url, additionalInfo)
    }
}
